core/secret.go

```diff
@@ -2,46 +2,21 @@ package core
 
 import (
 	"context"
-	"crypto/hmac"
-	"encoding/hex"
+	"fmt"
 	"sync"
 
 	"github.com/moby/buildkit/session/secrets"
 	"github.com/opencontainers/go-digest"
-	"github.com/pkg/errors"
 	"github.com/vektah/gqlparser/v2/ast"
 )
 
 // Secret is a content-addressed secret.
 type Secret struct {
 	Query *Query
 
-	// Name specifies the name of the secret.
-	Name string `json:"name,omitempty"`
-
-	// Accessor specifies the accessor key for the secret.
-	Accessor string `json:"accessor,omitempty"`
-}
-
-func GetLocalSecretAccessor(ctx context.Context, parent *Query, name string) (string, error) {
-	m, err := parent.CurrentModule(ctx)
-	if err != nil && !errors.Is(err, ErrNoCurrentModule) {
-		return "", err
-	}
-	var d digest.Digest
-	if m != nil {
-		d = m.Source.ID().Digest()
-	}
-	return NewSecretAccessor(name, d.String()), nil
-}
-
-func NewSecretAccessor(name string, scope string) string {
-	// Use an HMAC, which allows us to keep the scope secret
-	// This also protects from length-extension attacks (where if we had
-	// access to secret FOO in scope X, we could derive access to FOOBAR).
-	h := hmac.New(digest.SHA256.Hash, []byte(scope))
-	dt := h.Sum([]byte(name))
-	return hex.EncodeToString(dt)
+	// The digest of the DagQL ID that created this secret, used as its identifier
```

Comment:
nit: "Created" to me reads as the call that set the secret, but this is the ID of `secret(name: ...)`, not `setSecret`, which to me 'creates' the secret.

Maybe this?

```suggestion
	// The digest of the DagQL ID that accessed this secret, used as its identifier
```